Modules/Filtering/DistanceMap/include/itkDirectedHausdorffDistanceImageFilter.h

```diff
@@ -185,7 +185,7 @@ class ITK_TEMPLATE_EXPORT DirectedHausdorffDistanceImageFilter : public ImageToI
   DistanceMapPointer m_DistanceMap;
 
   Array<RealType>       m_MaxDistance;
-  Array<IdentifierType> m_PixelCount;
+  Array<IdentifierType> m_PixelCount = Array<RealType>::Fill(NumericTraits<IdentifierType>::ZeroValue());
```

Comment:
I don't understand this.  This is in the class declaration, for a non-static member so the initialization here is the default for any constructor that doesn't explicitly give an initialization for this member in the `: member0(value0), member1(value1)` part.  But the old code with the zero-number-of-arguments constructor did so with that constructor; so I don't understand why this is better.  Also, the right hand side is invoking `Fill` without a `this` pointer available, which I don't understand since `Fill` is not a static method.

Reply:
It appears that we are casting a pointer to the `Fill` method to be of type `Array<IdentifierType>::SizeValueType`.

Reply:
Right, calling `Fill` was not correct. The old code is producing warnings in the dynamic analysis. As said before, I ignore why since the array has no allocated size. Explicitly initialized to a zero size in ddd0e27.